src/lib/components/tooltip/QTooltipBase.svelte

```diff
@@ -0,0 +1,70 @@
+<script lang="ts">
+  import { onMount } from "svelte";
+  import { scale } from "svelte/transition";
+  import type { QTooltipProps } from "./props";
+
+  let {
+    // At this point, target should be guaranteed to be a DOM element
+    target,
+    position = "bottom",
+    offset = { x: 0, y: 0 },
+    children,
+    ...props
+  }: QTooltipProps<HTMLElement> = $props();
+
+  let tooltipEl = $state<HTMLDivElement>();
+
+  onMount(() => {
+    setTimeout(() => {
+      if (tooltipEl) {
+        tooltipEl.style.opacity = "1";
+      }
+    }, 50);
+  });
+
+  const tooltipPosition: Record<"top" | "left", number> | undefined = $derived.by(() => {
+    if (!target || !tooltipEl) {
+      return;
+    }
+
+    const anchorRect = target.getBoundingClientRect();
+
+    // 0.5rem offset so the tooltip is not stuck to the anchor
+    const defaultOffset = position.includes("-") ? 0 : 8;
+
+    const top = position.includes("top")
+      ? anchorRect.top - tooltipEl.offsetHeight + (offset.y || 0) - defaultOffset
+      : position.includes("bottom")
+        ? anchorRect.bottom + (offset.y || 0) + defaultOffset
+        : anchorRect.top + anchorRect.height / 2 - tooltipEl.offsetHeight / 2 + (offset.y || 0);
+
+    const left = position.includes("left")
+      ? anchorRect.left - tooltipEl.offsetWidth + (offset.x || 0) - defaultOffset
+      : position.includes("right")
+        ? anchorRect.right + (offset.x || 0) + defaultOffset
+        : anchorRect.left + anchorRect.width / 2 - tooltipEl.offsetWidth / 2 + (offset.x || 0);
+
+    return { top, left };
+  });
+
+  const styles = $derived(
+    tooltipPosition && { top: `${tooltipPosition.top}px`, left: `${tooltipPosition.left}px` }
```

Comment:
It's rather unusual to use `&&` to return a value. While you see `||` often, `&&` is not as easily understood for this by many. Maybe use a ternary instead (optional).

Reply:
Well, it's like writing `let activeClass = active && "my-class"`, it's exactly the same and we used it a lot so I think I'll keep it as is